app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -623,23 +696,42 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote, val w
 			}.setErrorCallback {
 				deferred.complete(LinkedList())
 			}
+		} else if (isArtistDirectory(directory)) {
```

Comment:
When do you encounter browseable Artists? The normal browse `Library > Artists > *` entries are only playable, not browseable for me.

Reply:
Browseable artists come from the search workflow, so if the user searches on a query and they select a search result that is an artist item (which is marked as browseable) then the user would be brought to the browse view to view that artist's songs which should be the top songs just like how the official Spotify experience is. I figured this would be a common flow for looking artists songs directly through the app. I don't have any Artists under the ```Library -> Artists -> * ``` part of the browse but I wonder if that is something that should be updated so it behaves in a similar fashion

Reply:
Indeed, the official Spotify experience does bring up a track listing when tapping into an Artist entry. It would be a decision between matching the original Android Auto browsing experience (playing immediately when clicking a browse entry for a specific Playlist/Album/Artist, helped by these recent improvements to Now Playing) or adding an extra browse layer. I'm currently leaning towards keeping the current state, with its intention to reduce decisions and distraction while driving.

Reply:
Ah I didn't know that the official Android Auto just plays browse entries like that, that's interesting. I can agree with leaving the browse in its current state then to minimize distractions and decisions while driving. I did notice that there is a slight issue with the queue that is created when playing an artist's playlist from the ```Library -> Artists -> *``` approach in that the queue metadata seems to be missing the title and there are entries for "Shuffle Play" and "Liked Songs" for some reason. Also there is the issue of being unable to play a specific song. I updated this workflow to use the Web API so now that playlist in the queue is working correctly.

I think this still makes the most sense for the search workflow perspective since that is already a pretty distracting workflow and the logic that already existed before these changes displayed the artist songs so my change here is to make it so that those artist songs are presented correctly and removing other possibly distracting information like weird playlist entries and other things.